src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1000,12 +1031,14 @@ private bool ImportIntrinsicCall(MethodDesc method)
 
             switch (method.Name)
             {
-                // Workaround for not being able to build a WASM version of CoreLib. This method
-                // would return the x64 size, which is too large for WASM
-                case "get_OffsetToStringData":
-                    if (metadataType.Name == "RuntimeHelpers" && metadataType.Namespace == "System.Runtime.CompilerServices")
+                case "get_Value":
+                    if (metadataType.IsByReferenceOfT)
                     {
-                        _stack.Push(new Int32ConstantEntry(8, _method.Context.GetWellKnownType(WellKnownType.Int32)));
+                        InstantiatedType instantiation = (InstantiatedType)metadataType;
```

Comment:
The cast to `InstantiatedType` is unnecessary as well. `Instantiation` is available on a `MetadataType` too.

Reply:
Fixed